tests/kokkos-based/overwriting_symmetric_matrix_vector_product.cpp

```diff
@@ -0,0 +1,134 @@
+
+#include "gtest_fixtures.hpp"
+#include "helpers.hpp"
+
+namespace
+{
+
+template<class A_t, class x_t, class y_t>
+void gemv_gold_solution(A_t A, x_t x, y_t y)
+{
+  for (std::size_t i=0; i<A.extent(0); ++i){
+    y(i) = typename y_t::value_type{};
+    for (std::size_t j=0; j<A.extent(1); ++j){
+      y(i) += A(i,j) * x(j);
+    }
+  }
+}
+
+template<class A_t, class Triangle, class x_t, class y_t>
+void kokkos_blas_overwriting_symv_impl(A_t A, Triangle t, x_t x, y_t y)
+{
+  namespace stdla = std::experimental::linalg;
+
+  using value_type = typename A_t::value_type;
+  assert(A.extent(0) == A.extent(1));
+  const std::size_t ext = A.extent(0);
+
+  // copy operands before running the kernel
+  auto A_preKernel = kokkostesting::create_stdvector_and_copy_rowwise(A);
+  auto x_preKernel = kokkostesting::create_stdvector_and_copy(x);
+  auto y_preKernel = kokkostesting::create_stdvector_and_copy(y);
+
+  // compute y gold symv
+  std::vector<value_type> gold(y.extent(0));
+  using mdspan_t = mdspan<value_type, extents<dynamic_extent>>;
+  mdspan_t y_gold(gold.data(), y.extent(0));
```

Comment:
This code is fine and doesn't need to change.  However, given that we're assuming C++20 anyway, one could use CTAD here:
```c++
mdspan y_gold(gold.data(), y.extent(0));
```
I'm not sure how far we want to back-port the reference implementation of P1673, though.

Reply:
I can leave this for now and revise them at the end ok? since these tests will end up in Kokkos kernels, not sure we want to restrict ourselves to c++20

Reply:
@fnrizzi Right, that's a good point -- it would be helpful if the tests didn't require C++20.